### PR TITLE
Temporary passwords fixes 31952

### PR DESCRIPTION
**Pull request by StCyr**
fixes to #31952 31952

### CODE REVIEW

File: apps/files_sharing/src/components/SharingEntryLink.vue [hunk 1]
```diff
@@ -192,6 +192,12 @@
 						@submit="onPasswordSubmit">
 						{{ t('files_sharing', 'Enter a password') }}
 					</ActionInput>
+					<ActionText v-if="passwordExpirationTime" icon="icon-info">
+						{{ t('files_sharing', 'Password expires {passwordExpirationTime}', {passwordExpirationTime}) }}
+					</ActionText>
+					<ActionText v-else-if="passwordExpirationTime !== null" icon="icon-error">
+						{{ t('files_sharing', 'Password expired') }}
```
Comment (PVince81) [inline apps/files_sharing/src/components/SharingEntryLink.vue:199]: when I set a new password, this message stays in place

expected: the message should switch to the above value "Password expires ..."

**Review by PVince81 — COMMENTED** [review]
Looks great, thanks a lot!

There's only one minor issue I found when the password already expired in the dropdown, see comment in the diff

**Comment by PVince81** [comment]
- [ ] BUG: when sharing by link (no email), the message "the password expires a few seconds ago" appears in the dropdown. Please add a condition check to not display this message at all for regular link shares

**Comment by StCyr** [comment]
There are conflicting files in the `dist` folder. Is it important @PVince81 ?

**Comment by StCyr** [comment]
fixed the wrongly shown "the password expires a few seconds ago" message in https://github.com/nextcloud/server/pull/31981/commits/4ac717b8a1880e93a0aa1ac526f98a6353d16901

**Comment by PVince81** [comment]
@StCyr the dist folder contains compiled/minified JS assets, and probably someone else changed something there on master.

I suggest to rebase your branch and ignore the "dist" files at first.
Then run "make build-js-production" locally (slow) and then force push.

If you forget the latter, we can also summon the bot with "/compile amend /" which will compile those assets for us.

**Comment by PVince81** [comment]
@StCyr please rebase onto master, this branch is still based on beta3.

also, did you address https://github.com/nextcloud/server/pull/31981#discussion_r871047464 ? (I haven't tested yet)

thanks!

**Comment by PVince81** [comment]
@StCyr you probably forgot to fetch or had an outdated master, this is why icons are missing (there was some work on master to fix it). Please try fetching again.

Or if you like I can take over

**Comment by StCyr** [comment]
> also, did you address [#31981 (comment)](https://github.com/nextcloud/server/pull/31981#discussion_r871047464) ? (I haven't tested yet)

Just did (in https://github.com/nextcloud/server/pull/31981/commits/9707bf459ea6449a54b593b37a0ef45d8335150d)

**Comment by Pytal** [comment]
`dist/icons.css` seems to be removed

**Comment by StCyr** [comment]
Seems good to go @PVince81

**Comment by StCyr** [comment]
No idea why it fails

**Comment by PVince81** [comment]
acceptance test failure likely unrelated, some of them fail sometimes

the node issue is known to fail on master: #32591

**Review by PVince81 — APPROVED** [review]
:+1:

@StCyr can you squash ?

File: apps/sharebymail/lib/ShareByMailProvider.php [hunk 2]
```diff
@@ -198,7 +198,7 @@ public function create(IShare $share) {
 
 		// Sends share password to receiver when it's a permanent one (otherwise she will have to request it via the showShare UI)
 		// or to owner when the password shall be given during a Talk session
-		if ($this->config->getSystemValue('sharing.enable_mail_link_password_expiration', false) === true || $share->getSendPasswordByTalk()) {
+		if ($this->config->getSystemValue('sharing.enable_mail_link_password_expiration', false) === false || $share->getSendPasswordByTalk()) {
```
Comment (artonge) [inline apps/sharebymail/lib/ShareByMailProvider.php:201]: Not sure about this change, could you explain it?

**Review by artonge — CHANGES_REQUESTED** [review]
Looks good overall. Remaining work:

- Rebase
- Drop `package-lock.json` changes
- Drop `icons.css` deletion
- Squash

File: apps/sharebymail/lib/ShareByMailProvider.php (hunk 2 above)
Comment (PVince81) [reply apps/sharebymail/lib/ShareByMailProvider.php:201]: @StCyr was it an error in logic that we missed the first time ?

Comment (StCyr) [reply apps/sharebymail/lib/ShareByMailProvider.php:201]: Yes, I think so. IIRC, we've inverted the setting enabling that feature -I'd initialy considered a setting more like "disable_mail_link_password_expiration"- and, it seems I forgot to change the logic here when doing this change.

The logic shall be:

Only send the share's password at share's creation when sharing.enable_mail_link_password_expiration is disabled.

Comment (artonge) [reply apps/sharebymail/lib/ShareByMailProvider.php:201]: So the settings name is reversed compared to how it is used ?

**Comment by StCyr** [comment]
Everything is green @CarlSchwan @artonge

File: apps/sharebymail/lib/ShareByMailProvider.php (hunk 2 above)
Comment (StCyr) [reply apps/sharebymail/lib/ShareByMailProvider.php:201]: >So the settings name is reversed compared to how it is used ?

when `sharing.enable_mail_link_password_expiration` is false, this means the instance doesn't use temporary passwords for mail shares and thus that the password must be sent to the recipient.

The setting's name is correct: it's meant to tell whether temporary passwords are enabled on the instance. It is not meant to tell whether the passwords shall be sent by mail or not.

**Review by artonge — APPROVED** [review]

**Pull request merged**